### PR TITLE
Add binary motion sensor (Second Try)

### PR DESCRIPTION
**Pull request by DanielHabenicht**
implements the features from #64

Fixes the Following error which occurred on initialization and was the reason for the ever #71:

image

### CODE REVIEW

**Comment by planbnet** [comment]
Yes, it was something like that. off_delay could not be found on initial startup. After that, I set the off delay on the device, but the sensor still did not turn on or off afterwards, event after a reload.

**Comment by planbnet** [comment]
I'll test this on this weekend, but I'll wait for 1.9.2 to settle a bit before merging this and the button pr (and doing a 1.10.0 release) so we start from a stable state. I did quite a few changes and I want the integration to be stable for users when livisi turns off their servers on march 1st.

**Comment by DanielHabenicht** [comment]
sure. If you want you can test again. I (hopefully) fixed the initialization bug with the last commit.

**Comment by planbnet** [comment]
Does not seem to be fixed yet:
```
2024-02-23 09:01:03.891 DEBUG (MainThread) [custom_components.livisi] {"type": "MotionDetected","namespace": "CosipDevices.RWE","desc": "/desc/device/WMDO.RWE/1.0/capability/MotionDetectionSensor/event/MotionDetected","source": "/capability/2aac7491a45a441cb8014cf5ef3c6df2","timestamp": "2024-02-23T09:01:03.873074Z","properties": {"luminance": 36},"context": {"details": {}}}
2024-02-23 09:01:03.896 ERROR (MainThread) [homeassistant.util.logging] Exception in trigger_event when dispatching 'livisi_event_2aac7491a45a441cb8014cf5ef3c6df2': ({'device_id': '99946cc7639a484fa94815c22ec9ecb3', 'type': 'motion_detected'},)
Traceback (most recent call last):
  File "/workspaces/livisi_unofficial/custom_components/livisi/binary_sensor.py", line 282, in trigger_event
    off_delay = self.get_off_delay()
                ^^^^^^^^^^^^^^^^^^^^
  File "/workspaces/livisi_unofficial/custom_components/livisi/binary_sensor.py", line 300, in get_off_delay
    return float(state.state)
                 ^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'state'
```

**Comment by DanielHabenicht** [comment]
> Does not seem to be fixed yet:
> 
> ```
> 2024-02-23 09:01:03.891 DEBUG (MainThread) [custom_components.livisi] {"type": "MotionDetected","namespace": "CosipDevices.RWE","desc": "/desc/device/WMDO.RWE/1.0/capability/MotionDetectionSensor/event/MotionDetected","source": "/capability/2aac7491a45a441cb8014cf5ef3c6df2","timestamp": "2024-02-23T09:01:03.873074Z","properties": {"luminance": 36},"context": {"details": {}}}
> 2024-02-23 09:01:03.896 ERROR (MainThread) [homeassistant.util.logging] Exception in trigger_event when dispatching 'livisi_event_2aac7491a45a441cb8014cf5ef3c6df2': ({'device_id': '99946cc7639a484fa94815c22ec9ecb3', 'type': 'motion_detected'},)
> Traceback (most recent call last):
>   File "/workspaces/livisi_unofficial/custom_components/livisi/binary_sensor.py", line 282, in trigger_event
>     off_delay = self.get_off_delay()
>                 ^^^^^^^^^^^^^^^^^^^^
>   File "/workspaces/livisi_unofficial/custom_components/livisi/binary_sensor.py", line 300, in get_off_delay
>     return float(state.state)
>                  ^^^^^^^^^^^
> AttributeError: 'NoneType' object has no attribute 'state'
> ```

When did that happen? On a fresh install?

**Comment by planbnet** [comment]
Updated from 1.9.2, the error came after the motion detector detected motion for the first time.

**Comment by planbnet** [comment]
Sorry I broke mergeability of this PR by renaming `async_get_device_state` to `async_get_value` and creating `async_get_state` which returns the full dict of properties. You can simply use async_get_state().get("lastChanged") now. This might be useful later, if multiple properties are required (we don't want to send multiple requests to the same endpoint ins this case).

**Comment by planbnet** [comment]
I think I found the problem:  `self.off_delay_entity_id: str = Platform.NUMBER + "." + device.name+ "_duration"`, 
`self.hass.states.get(id)` is looking for "number.Bewegungsmelder (Außen)_duration" in my case which obviously causes problems...

File: custom_components/livisi/number.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        self.entity_description = entity_desc
+        unique_id = device.id + "_" + entity_desc.key + "_number"
+        self._attr_unique_id = unique_id
+        self.device_id = device.id
+        self._attr_translation_key = entity_desc.key
+        self.entity_id = str.lower(
+            Platform.NUMBER + "." + device.id + "_" + entity_desc.key
+        )
+
+        self._attr_device_info = DeviceInfo(
```
Comment (planbnet) [inline custom_components/livisi/number.py:92]: I don't like that this is duplicated code from the base class... If anything changes there, we will definitely forget changing it here too...

Comment (planbnet) [reply custom_components/livisi/number.py]: Fixed by creating create_device_info

File: custom_components/livisi/binary_sensor.py [hunk 2]
```diff
@@ -177,3 +199,126 @@ def _handle_coordinator_update(self) -> None:
...
+            self._delay_listener = evt.async_call_later(
+                self.hass, off_delay, off_delay_listener
+            )
+
+    def get_off_delay(self) -> float:
+        """Get the Delay."""
+        id = self.off_delay_unique_id
+
+        entity_registry = er.async_get(self.hass)
+        entity_id = entity_registry.entities.get_entity_id(
```
Comment (planbnet) [inline custom_components/livisi/binary_sensor.py:314]: Using a unique id and retrieving the config item via it (and not via the changeable entity id) should fix the problem...

**Comment by DanielHabenicht** [comment]
Currently on vacation, so i wont be able to test for 2 weeks

**Comment by planbnet** [comment]
Currently, hassfest validation of the translation files fails:
```
Integration livisi - /github/workspace/custom_components/livisi:
Error: R] [JSON] Invalid JSON file translations/en.json
Error: R] [TRANSLATIONS] Invalid JSON in translations/en.json: Extra data: line 91 column 1 (char 2975)
Warning: G] [MANIFEST] Domain collides with built-in core integration
```

I guess the key cannot be named "duration"

**Comment by DanielHabenicht** [comment]
Yeah that was working and not working when I tried it. Because it should be translated from the core package but when I tried it did not work.

**Comment by DanielHabenicht** [comment]
works for me. Seems like hassfest validator is working again?

**Pull request merged**